include/picongpu/particles/debyeLength/Estimate.kernel

```diff
@@ -118,28 +118,24 @@ namespace picongpu
                     auto sumWeighting = float_64{0.0};
                     auto sumMomentum = float3_64::create(0.0);
                     auto sumMomentumSquared = float3_64::create(0.0);
-                    auto frame = electronBox.getLastFrame(supercellIdx);
-                    auto particlesInFrame = electronBox.getSuperCell(supercellIdx).getSizeLastFrame();
+
                     uint32_t const workerIdx = cupla::threadIdx(acc).x;
-                    while(frame.isValid())
-                    {
-                        lockstep::makeForEach<frameSize, numWorkers>(workerIdx)(
-                            [&](uint32_t const linearIdx)
-                            {
-                                if(linearIdx < particlesInFrame)
-                                {
-                                    auto const particle = frame[linearIdx];
-                                    auto const weighting = float_64{particle[weighting_]};
-                                    sumWeighting += weighting;
-                                    auto const singleParticleMomentum
-                                        = precisionCast<float_64>(particle[momentum_]) / weighting;
-                                    sumMomentum += weighting * singleParticleMomentum;
-                                    sumMomentumSquared += weighting * singleParticleMomentum * singleParticleMomentum;
-                                }
-                            });
-                        frame = electronBox.getPreviousFrame(frame);
-                        particlesInFrame = frameSize;
-                    }
+                    auto forEachParticle = pmacc::particles::algorithm::acc::makeForEach<numWorkers>(
+                        workerIdx,
+                        electronBox,
+                        supercellIdx);
```

Comment:
This is only possible if forEachParticle is rewritten to work with runtime sizes only. The number of workers depends on the alpaka accelerator we are using. I would avoid using any `default numWorkers` because the default will never be good. 
A forEach which is fully runtime would IMO make more sense.

Reply:
What i meant is, most PIConGPU particle-based kernels (current deposition is exception) use template parameter `uint32_t T_numWorkers` and on the host side we use `pmacc::traits::GetNumWorkers<...>`. I was thinking to just simplify this common case by using this value by default.

Reply:
Discussed offline, as a follow-up to this work we can make some helpers to simplify writing and launching such kernels. That may also include ensuring that the `T_numWorkers` (or a new way to express it) matches the kernel block size - which is a requirement even now, but we do not check for it.